tests/integration/test_dev_server.py

```diff
@@ -0,0 +1,156 @@
+import asyncio
+import nikola.plugins.command.auto as auto
+from nikola.utils import get_logger
+import pytest
+import pathlib
+import socket
+from typing import Optional
+import urllib.request
+
+from ..helper import FakeSite
+
+SERVER_ADDRESS = "localhost"
+TEST_MAX_DURATION = 10  # Watchdog: Give up the test if it did not succeed during this time span.
+SERVER_STARTUP_TIME = 0.1  # Max time, in seconds, we expect the server to need to come up.
+
+# Folder that has the fixture file we expect the server to serve:
+OUTPUT_FOLDER = pathlib.Path.cwd() / "tests" / "data" / "dev_server_sample_output_folder"
```

Comment:
Depending on the current working directory is flaky and may break. Instead, start with `__file__` (path to the current `.py` file) and find the folder based on that.

Reply:
I'd actually speculate `pytest tests/` is stable and the place of the test source in the directory tree is less stable. But it's not worth investing much thought into. So as you please.